src/Vault.sol

```diff
@@ -0,0 +1,398 @@
+        return _getStrategyStorage().strategies[asset_];
+    }
+
+    function previewDepositAsset(address asset_, uint256 assets_) public view returns (uint256) {
+        if (!_getAssetStorage().assets[asset_].active) revert InvalidAsset();
+        return _convertToShares(asset_, assets_, Math.Rounding.Floor);
+    }
+
+    function depositAsset(address asset_, uint256 assets_, address receiver) public nonReentrant returns (uint256) {
+        if (paused()) revert Paused();
+        if (!_getAssetStorage().assets[asset_].active) revert InvalidAsset();
+
+        uint256 shares = previewDepositAsset(asset_, assets_);
+        _deposit(asset_, _msgSender(), receiver, assets_, shares);
+
+        return shares;
+    }
+
+    function paused() public view returns (bool) {
+        return _getVaultStorage().paused;
+    }
+
+    function rateProvider() public view returns (address) {
+        return _getVaultStorage().rateProvider;
+    }
+
+    function bufferStrategy() public view returns (address) {
+        return _getVaultStorage().bufferStrategy;
+    }
+
+    function processAccounting() public {
+        AssetStorage storage assetStorage = _getAssetStorage();
+        StrategyStorage storage strategyStorage = _getStrategyStorage();
+        uint256 totalBaseBalance = 0;
+
+        for (uint256 i = 0; i < assetStorage.list.length; i++) {
+            address asset_ = assetStorage.list[i];
+            uint256 idleBalance = IERC20(asset_).balanceOf(address(this));
+            assetStorage.assets[asset_].idleBalance = idleBalance;
+            totalBaseBalance += _convertAssetToBase(asset_, idleBalance);
+        }
+
+        for (uint256 i = 0; i < strategyStorage.list.length; i++) {
+            address strategy = strategyStorage.list[i];
+            uint256 idleBalance = IERC20(strategy).balanceOf(address(this));
+            strategyStorage.strategies[strategy].idleBalance = idleBalance;
+            totalBaseBalance += _convertAssetToBase(strategy, idleBalance);
+        }
+
+        _getVaultStorage().totalAssets = totalBaseBalance;
+    }
+
+    function processAllocation(address[] calldata targets, uint256[] memory values, bytes[] calldata data)
+        external
+        onlyRole(DEFAULT_ADMIN_ROLE)
+    {
+        for (uint256 i = 0; i < targets.length; i++) {
+            bool success;
+            bytes memory returnData;
+
+            // Determine the type of target and process accordingly
+            if (_getAssetStorage().assets[targets[i]].active) {
+                // Process asset approvals // QUESTION: Assumiung we only need approvals for assets
+                if (data[i].length > 0 && data[i][0] == abi.encodeWithSignature("approve(address,uint256)")[0]) {
+                    (success, returnData) = targets[i].call{value: values[i]}(data[i]);
+                    if (!success) revert ProcessFailed(returnData);
+                }
+            } else if (_getStrategyStorage().strategies[targets[i]].active) {
```

Comment:
this doesn't work for handling wihdrawals on ynETH/ynLSDe currently as you interact with a different contract.

We can potentially consider to upgrade those to ebmed withdrawals into the token assets, but even then not all strategies support calls targeted at a single contract (eg. etherfi WEETH)

Reply:
yeah fixed this, see commented above